File: python/7 kyu/371.py
# https://www.codewars.com/kata/546ba103f0cf8f7982000df4/train/python

# Passed

def calculate(n1, n2, o):
    operations = {
        'add': lambda decimal_num_1, decimal_num_2: decimal_num_1 + decimal_num_2,
        'multiply': lambda decimal_num_1, decimal_num_2: decimal_num_1 * decimal_num_2,
        'subtract': lambda decimal_num_1, decimal_num_2: decimal_num_1 - decimal_num_2
    }
    
    operation = operations[o]
    binary_result = bin(operation(int(n1, 2), int(n2, 2)))
    sign, binary = binary_result.split('b')
    result = ('' if sign == '0' else '-') + binary
    
    return result

output = calculate('1', '1', 'add')
print(output)

File: nodejs/7 kyu/371.js
// https://www.codewars.com/kata/58a66c208b88b2de660000c3/train/javascript

// Passed

function missingValues(arr) {
  let numToCountObj = new Object();
  for (const num of arr) {
    if (!(num in numToCountObj)) {
      numToCountObj[num] = 0;
    }

    numToCountObj[num]++;
  }

  let x, y;
  for (const num of Object.keys(numToCountObj)) {
    let count = numToCountObj[num];
    if (count === 1) {
      x = num;
    } else if (count === 2) {
      y = num;
    }
  }

  let result = x * x * y;
  return result;
}

const output = missingValues([1, 1, 1, 2, 2, 3]);
console.log(output);